R/lazyframe-frame.R

```diff
@@ -2361,4 +2361,199 @@ lazyframe__join_asof <- function(
   })
 }
 
+#' Creates a summary of statistics for a LazyFrame, returning a DataFrame.
+#'
+#' @description
+#' This method does not maintain the laziness of the frame, and will collect
+#' the final result. This could potentially be an expensive operation.
+#'
+#' We do not guarantee the output of `describe()` to be stable. It will show
+#' statistics that we deem informative, and may be updated in the future. Using
+#' `describe()` programmatically (versus interactive exploration) is not
+#' recommended for this reason.
+#'
+#' @param percentiles One or more percentiles to include in the summary
+#' statistics. All values must be in the range `[0; 1]`.
+#' @param interpolation Interpolation method for computing quantiles. Must be
+#' one of `"nearest"`, `"higher"`, `"lower"`, `"midpoint"`, or `"linear"`.
+#'
+#' @details
+#' The median is included by default as the 50% percentile.
+#'
+#' @inherit as_polars_df return
+#' @examples
+#' lf <- pl$LazyFrame(
+#'   int = 1:3,
+#'   float = c(0.5, NA, 2.5),
+#'   string = c(letters[1:2], NA),
+#'   date = c(as.Date("2024-01-20"), as.Date("2024-01-21"), NA),
+#'   cat = factor(c(letters[1:2], NA)),
+#'   bool = c(TRUE, FALSE, NA)
+#' )
+#' lf$collect()
+#'
+#' # Show default frame statistics:
+#' lf$describe()
+#'
+#' # Customize which percentiles are displayed, applying linear interpolation:
+#' lf$describe(
+#'   percentiles = c(0.1, 0.3, 0.5, 0.7, 0.9),
+#'   interpolation = "linear"
+#' )
+lazyframe__describe <- function(
+  percentiles = c(0.25, 0.5, 0.75),
+  interpolation = c("nearest", "higher", "lower", "midpoint", "linear")
+) {
```

Comment:
In Python Polars, `interpolation` must be named.
So we need `...` and check dots empty.

Reply:
Good catch, done